src/cli/api/apihelper.go

```diff
@@ -0,0 +1,381 @@
+
+	err := helper.CheckHealth()
+	if err != nil {
+		return err
+	}
+
+	baseURL := helper.Endpoint.URL
+	requestURL := fmt.Sprintf("%s%s", baseURL, strings.Replace(PolicyPath, "{appId}", helper.Client.AppId, -1))
+
+	var body io.Reader
+	if data != nil {
+		jsonByte, e := json.Marshal(data)
+		if e != nil {
+			return fmt.Errorf(ui.InvalidPolicy, e)
+		}
+		body = bytes.NewBuffer(jsonByte)
+	}
+
+	req, err := http.NewRequest("PUT", requestURL, body)
+	req.Header.Add("Authorization", helper.Client.AuthToken)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := helper.DoRequest(req, baseURL)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	raw, err := ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+
+		var errorMsg string
+		switch resp.StatusCode {
+		case 401:
+			errorMsg = fmt.Sprintf(ui.Unauthorized, baseURL, helper.Client.CCAPIEndpoint)
+		case 400:
+			errorMsg, err = parseErrResponse(raw)
+			if err != nil {
+				return err
+			}
+			errorMsg = fmt.Sprintf(ui.InvalidPolicy, errorMsg)
+
+		default:
+			errorMsg, err = parseErrResponse(raw)
+			if err != nil {
+				return err
+			}
+		}
+		return errors.New(errorMsg)
+	}
+	return nil
+}
+
+func (helper *APIHelper) DeletePolicy() error {
+
+	err := helper.CheckHealth()
+	if err != nil {
+		return err
+	}
+
+	baseURL := helper.Endpoint.URL
+	requestURL := fmt.Sprintf("%s%s", baseURL, strings.Replace(PolicyPath, "{appId}", helper.Client.AppId, -1))
+
+	req, err := http.NewRequest("DELETE", requestURL, nil)
+	req.Header.Add("Authorization", helper.Client.AuthToken)
+
+	resp, err := helper.DoRequest(req, baseURL)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	raw, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		var errorMsg string
+		switch resp.StatusCode {
+		case 401:
+			errorMsg = fmt.Sprintf(ui.Unauthorized, baseURL, helper.Client.CCAPIEndpoint)
+		case 404:
+			errorMsg = fmt.Sprintf(ui.PolicyNotFound, helper.Client.AppName)
+		default:
+			errorMsg, err = parseErrResponse(raw)
+			if err != nil {
+				return err
+			}
+		}
+		return errors.New(errorMsg)
+	}
+
+	return nil
+
+}
+
+func (helper *APIHelper) GetMetrics(metricName string, startTime, endTime int64, desc bool, page uint64) (bool, [][]string, error) {
+
+	next := false
+	if page <= 1 {
+		err := helper.CheckHealth()
+		if err != nil {
+			return next, nil, err
+		}
+	}
+
+	baseURL := helper.Endpoint.URL
+	queryMetricURL := strings.Replace(MetricPath, "{appId}", helper.Client.AppId, -1)
+	queryMetricURL = strings.Replace(queryMetricURL, "{metric_type}", metricName, -1)
+	requestURL := fmt.Sprintf("%s%s", baseURL, queryMetricURL)
+
+	req, err := http.NewRequest("GET", requestURL, nil)
+	req.Header.Add("Authorization", helper.Client.AuthToken)
+	q := req.URL.Query()
+	if startTime != 0 {
+		q.Add("start-time", strconv.FormatInt(startTime, 10))
+	}
+	if endTime != 0 {
```

Comment:
if endTime is not specified,  it means NOW.. so this is not correct.

Reply:
and startTime and endTime needs to be >=0

Reply:
it is a validation step for the query parameters.  I changed them >0  now to  avoid invalid input